pkg/manifest/writer/manifest_writer.go

```diff
@@ -185,15 +185,19 @@ func toWriteableURL(url manifest.URLDefinition) persistence.TypedValue {
 }
 
 // getTokenSecret returns the tokenConfig with some legacy magic string append that still might be used (?)
-func getTokenSecret(a manifest.Auth, envName string) persistence.AuthSecret {
+func getTokenSecret(a manifest.Auth, envName string) *persistence.AuthSecret {
+	if a.Token == nil {
+		return nil
+	}
+
 	var envVarName string
```

Comment:
since you've touched that function, lets clean it up a bit:
```
func getTokenSecret(a manifest.Auth, envName string) *persistence.AuthSecret {
	if a.Token == nil {
		return nil
	}

	envVarName := a.Token.Name
	if envVarName == "" {
		envVarName = envName + "_TOKEN"
	}

	return &persistence.AuthSecret{
		Type: persistence.TypeEnvironment,
		Name: envVarName,
	}
}
```